backend/src/main/twirl/MainIOVertx.scala.txt

```diff
@@ -0,0 +1,52 @@
+@(groupId: String, addDocumentation: Boolean, addMetrics: Boolean)
+package @groupId
+
+import cats.effect._
+import cats.effect.std.Dispatcher
+import io.vertx.core.Vertx
+import io.vertx.ext.web.Router
+import sttp.tapir.server.vertx.cats.@if(addMetrics){{VertxCatsServerInterpreter, VertxCatsServerOptions}}else{VertxCatsServerInterpreter}
+import sttp.tapir.server.vertx.cats.VertxCatsServerInterpreter._
+
+import scala.io.StdIn
+
+
+object Main extends IOApp {
+
+  override def run(args: List[String]): IO[ExitCode] = {
+    val port = sys.env.get("http.port").map(_.toInt).getOrElse(8080)
+
+    val vertx = Vertx.vertx()
+    val server = vertx.createHttpServer()
+    val router = Router.router(vertx)
+
+    Dispatcher[IO]
+      @if(addMetrics) {
+      .map(d => {
+        VertxCatsServerOptions
+          .customiseInterceptors(d)
+          .metricsInterceptor(Endpoints.prometheusMetrics.metricsInterceptor())
+          .options
+      })
+      }
+      .use { @if(addMetrics) {serverOptions} else {dispatcher} =>
+        for {
+          bind <- IO.delay {
+            Endpoints.all
+              .foreach(endpoint => {
+                VertxCatsServerInterpreter[IO](@if(addMetrics) {serverOptions} else {dispatcher})
+                  .route(endpoint)
+                  .apply(router)
+              })
+            server.requestHandler(router).listen(port)
+          }
+          _ <- IO.blocking {
+            println(s"@if(addDocumentation){Go to http://localhost:${port}/docs to open SwaggerUI. }else{Server started at http://localhost:${port}. }Press ENTER key to exit.")
```

Comment:
there `${port}` but here and in the following templates the actual port (either from `bind` or form the server instance) needs to be returned as tests pass `0` so that server could pick-up the random port and that random port needs to be printed so that client can read it from logs and connect to it :)